integrations/google_genai/src/haystack_integrations/components/generators/google_genai/chat/chat_generator.py

```diff
@@ -0,0 +1,579 @@
+                types.Part(
+                    function_call=types.FunctionCall(
+                        id=tool_call.id, name=tool_call.tool_name, args=tool_call.arguments
+                    )
+                )
+            )
+
+    # Handle tool call results (from tool/user)
+    if message.tool_call_results:
+        for result in message.tool_call_results:
+            parts.append(
+                types.Part(
+                    function_response=types.FunctionResponse(
+                        id=result.origin.id, name=result.origin.tool_name, response={"result": result.result}
+                    )
+                )
+            )
+
+    # Determine role
+    if message.is_from(ChatRole.USER) or message.tool_call_results:
+        role = "user"
+    elif message.is_from(ChatRole.ASSISTANT):
+        role = "model"
+    elif message.is_from(ChatRole.SYSTEM):
+        # System messages will be handled separately as system instruction
+        role = "user"
+    else:
+        msg = f"Unsupported message role: {message._role}"
+        raise ValueError(msg)
+
+    return types.Content(role=role, parts=parts)
+
+
+def _sanitize_tool_schema(tool_schema: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Sanitizes a tool schema to remove any keys that are not supported by Google Gen AI.
+
+    Google Gen AI does not support additionalProperties, $defs, or $ref in the tool schema.
+
+    :param tool_schema: The tool schema to sanitize.
+    :returns: The sanitized tool schema.
+    """
+    # google Gemini does not support additionalProperties in the tool schema
+    tool_schema = remove_key_from_schema(tool_schema, "additionalProperties")
+    # expand $refs in the tool schema
+    tool_schema = replace_refs(tool_schema)
+    # and remove the $defs key leaving the rest of the schema
+    tool_schema = remove_key_from_schema(tool_schema, "$defs")
+
+    return tool_schema
+
+
+def _convert_tools_to_google_genai_format(tools: Union[List[Tool], Toolset]) -> List[types.Tool]:
+    """
+    Converts a list of Haystack Tools or a Toolset to Google Gen AI Tool format.
+
+    :param tools: List of Haystack Tool objects or a Toolset.
+    :returns: List of Google Gen AI Tool objects.
+    """
+    # Convert Toolset to list if needed
+    if isinstance(tools, Toolset):
+        tools = list(tools)
+
+    function_declarations = []
+    for tool in tools:
+        parameters = _sanitize_tool_schema(tool.parameters)
+        function_declarations.append({"name": tool.name, "description": tool.description, "parameters": parameters})
+
+    # Return a single Tool object with all function declarations as in the Google GenAI docs
+    # we could also return multiple Tool objects, doesn't seem to make a difference
+    # revisit this decision
+    return [types.Tool(function_declarations=function_declarations)]  # type: ignore[arg-type]
+
+
+def _convert_google_genai_response_to_chatmessage(response, model: str) -> ChatMessage:
```

Comment:
Could we add the type hint here?
```suggestion
def _convert_google_genai_response_to_chatmessage(response: types.GenerateContentResponse, model: str) -> ChatMessage:
```